src/masternodes/rpc_vault.cpp

```diff
@@ -249,9 +247,13 @@ namespace {
             } else {
                 interestsPerBlockBalances = AmountsToJSON(interestsPerBlock);
                 totalInterestsPerBlockValue = ValueFromAmount(totalInterestsPerBlock);
+            }
+            if (!isVaultTokenLocked){
+                // IPB can only be calculated when tokens are not locked.
+                // Value during split period is hard to predict with the current|next price
+                result.pushKV("interestsPerBlock", interestsPerBlockBalances);
```

Comment:
Looks good, but still keep the json keys and set it to -1. Don't skip the values.